.travis.yml

```diff
@@ -1,32 +1,80 @@
-os: linux
-dist: trusty
-sudo: required
-language: python
-python:
-  - "2.7"
-  - "3.6"
-
-addons:
-  apt:
-    packages:
-      - dos2unix
+language: generic
+
+matrix:
+  include:
+    - os: linux
+      sudo: required
+      env: PYTHON_VERSION=python2
+    - os: linux
+      sudo: required
+      env: PYTHON_VERSION=python3
+    - os: osx
+      osx_image: xcode9.3beta
+      env: PYTHON_VERSION=python2
+    - os: osx
+      osx_image: xcode9.3beta
+      env: PYTHON_VERSION=python3
+    - os: linux
+      sudo: required
+      env: PYTHON_VERSION=python2 USE_NINJA=true
+    - os: linux
+      sudo: required
+      env: PYTHON_VERSION=python3 USE_NINJA=true
+    - os: osx
+      osx_image: xcode9.3beta
+      env: PYTHON_VERSION=python2 USE_NINJA=true
+    - os: osx
+      osx_image: xcode9.3beta
+      env: PYTHON_VERSION=python3 USE_NINJA=true
+    - os: linux
+      sudo: required
+      env: PYTHON_VERSION=python2 ONNX_ML=1
+    - os: linux
+      sudo: required
+      env: PYTHON_VERSION=python3 ONNX_ML=1
+    - os: osx
+      osx_image: xcode9.3beta
+      env: PYTHON_VERSION=python2 ONNX_ML=1
+    - os: osx
+      osx_image: xcode9.3beta
+      env: PYTHON_VERSION=python3 ONNX_ML=1
+    - os: linux
+      sudo: required
+      env: PYTHON_VERSION=python2 USE_NINJA=true ONNX_ML=1
+    - os: linux
+      sudo: required
+      env: PYTHON_VERSION=python3 USE_NINJA=true ONNX_ML=1
+    - os: osx
+      osx_image: xcode9.3beta
+      env: PYTHON_VERSION=python2 USE_NINJA=true ONNX_ML=1
+    - os: osx
+      osx_image: xcode9.3beta
+      env: PYTHON_VERSION=python3 USE_NINJA=true ONNX_ML=1
+
+env:
+  global:
+    - BUILD_CCACHE_DIR=~/build/ccache
+    - BUILD_NINJA_DIR=~/build/ninja
+
+before_install:
+  - ./.travis/before_install.sh
 
 install:
   - ./.travis/install.sh
 
 script:
-  - ./.travis/build.sh
+  - ./.travis/script.sh
+
+after_success:
+  - ./.travis/after_success.sh
+
+after_failure:
+  - ./.travis/after_failure.sh
 
 cache:
+  - timeout: 300
   - directories:
-    - $HOME/.build_cache
+    - $BUILD_CCACHE_DIR
+    - $BUILD_NINJA_DIR
     - $HOME/.ccache
-
-env:
-  global:
-    - PB_VERSION=2.6.1
```

Comment:
We need this. travis builds use environment variables to separate build caches.

Reply:
++ should we be also setting it in onnx-coreml then?

Reply:
I'm on it